protocols/imap/src/main/java/org/apache/james/imap/decode/parser/GetAnnotationCommandParser.java

```diff
@@ -29,119 +28,99 @@
+    private final static String MAXSIZE = "MAXSIZE";
+    private final static String DEPTH = "DEPTH";
 
     public GetAnnotationCommandParser() {
         super(ImapCommand.authenticatedStateCommand(ImapConstants.GETANNOTATION_COMMAND_NAME));
     }
 
     @Override
-    protected ImapMessage decode(ImapCommand command, ImapRequestLineReader request, String tag, ImapSession session)
+    protected ImapMessage decode(ImapCommand command, ImapRequestLineReader requestReader, String tag, ImapSession session)
             throws DecodingException {
+        GetAnnotationRequest.GetAnnotationRequestBuilder builder = new GetAnnotationRequest.GetAnnotationRequestBuilder(tag, command);
         try {
-            GetAnnotationRequest annotationRequest = new GetAnnotationRequest(tag, command, request.mailbox());
+            builder.mailboxName(requestReader.mailbox());
 
-            while (nextNonSpaceChar(request) == '(') {
-                consumeParenthesis(request, annotationRequest);
+            while (requestReader.nextNonSpaceChar() == '(') {
+                consumeAnnotationRequestProperty(requestReader, builder);
             }
 
-            if (ENDOFLINE_PATTERN.matches(nextNonSpaceChar(request))) {
-                consumeKey(request, annotationRequest);
+            if (ENDOFLINE_PATTERN.matches(requestReader.nextNonSpaceChar())) {
+                consumeKey(requestReader, builder);
             }
 
-            Preconditions.checkArgument(annotationRequest.isValidParams());
-            return annotationRequest;
+            return builder.build();
         } catch (NullPointerException e) {
             throw new DecodingException(HumanReadableText.ILLEGAL_ARGUMENTS, e.getMessage());
         } catch (IllegalArgumentException e) {
             throw new DecodingException(HumanReadableText.ILLEGAL_ARGUMENTS, e.getMessage());
         }
     }
 
-    private void consumeParenthesis(ImapRequestLineReader request, GetAnnotationRequest getAnnotationRequest) throws DecodingException {
-        request.consumeChar('(');
-        switch (request.nextChar()) {
+    private void consumeAnnotationRequestProperty(ImapRequestLineReader requestReader, GetAnnotationRequest.GetAnnotationRequestBuilder builder) throws DecodingException {
+        requestReader.consumeChar('(');
+        boolean shouldEndOfLine = false;
+        switch (requestReader.nextChar()) {
             case 'M':
-                getAnnotationRequest.setMaxsizeValue(consumeMaxsizeOpt(request));
+                builder.maxsize(consumeMaxsizeOpt(requestReader));
                 break;
 
             case 'D':
-                getAnnotationRequest.setDepthValue(consumeDepthOpt(request));
+                builder.depth(consumeDepthOpt(requestReader));
                 break;
 
             default:
-                getAnnotationRequest.setKeys(consumeKeys(request));
+                builder.keys(consumeKeys(requestReader));
+                shouldEndOfLine = true;
```

Comment:
You should not modify the sate of the method in the switch. 
Switch instruction are hard to read and understand if more than 1 line.